siliconcompiler/tools/vpr/route.py

```diff
@@ -0,0 +1,84 @@
+import os
+import re
+import shutil
+
+from siliconcompiler.tools.vpr import vpr
+
+
+def setup(chip):
+    '''
+    Perform automated place and route with VPR
+    '''
+
+    tool = 'vpr'
+    step = chip.get('arg', 'step')
+    index = chip.get('arg', 'index')
+    task = chip._get_task(step, index)
+
+    chip.set('tool', tool, 'exe', tool, clobber=False)
+    chip.set('tool', tool, 'version', '0.0', clobber=False)
+
+    chip.set('tool', tool, 'task', task, 'threads', os.cpu_count(),
+             step=step, index=index, clobber=False)
+
+    # TO-DO: PRIOROTIZE the post-routing packing results?
+    design = chip.top()
+    chip.add('tool', tool, 'task', task, 'output', design + '.route', step=step, index=index)
+    chip.add('tool', tool, 'task', task, 'output', 'vpr_stdout.log', step=step, index=index)
+
+    options = vpr.assemble_options(chip, tool)
+    # Confine VPR execution to routing step
+    options.append('--route')
+    # To run only the routing step we need to pass in the placement files
+    options.append(f'--net_file inputs/{design}.net')
+    options.append(f'--place_file inputs/{design}.place')
+
+    threads = chip.get('tool', tool, 'task', task, 'threads', step=step, index=index)
+    options.append(f"--num_workers {threads}")
+
+    chip.add('tool', tool, 'task', task, 'option', options, step=step, index=index)
+
+#############################################
+# Runtime pre processing
+#############################################
+
+
+def pre_process(chip):
+
+    # have to rename the net connected to unhooked pins from $undef to unconn
+    # as VPR uses unconn keywords to identify unconnected pins
+
+    step = chip.get('arg', 'step')
+    index = chip.get('arg', 'index')
+    design = chip.top()
+    blif_file = f"{chip._getworkdir()}/{step}/{index}/inputs/{design}.blif"
+    print(blif_file)
+    with open(blif_file, 'r+') as f:
+        netlist = f.read()
+        f.seek(0)
+        netlist = re.sub(r'\$undef', 'unconn', netlist)
+        f.write(netlist)
+        f.truncate()
```

Comment:
why are you rewriting the input file? If this is needed, you should instead rename the input file something like `sc_{design}.blif` since we use hardlinks for the input files, which means you are actually overwriting the output of the previous step. and up the flow to point to the `sc_{}.blif` file instead.

Reply:
This was inherited from previous development efforts.  Removing it.